src/dendron/actions/generate.py

```diff
@@ -0,0 +1,130 @@
+from dendron.action_node import ActionNode
+from dendron.basic_types import NodeStatus
+from dendron.actions.configs.hflm_config import HFLMActionConfig
+
+from typing import Callable
+
+import types
+from hflm.huggingface_model import HFLM
+import traceback
+
+class GenerateAction(ActionNode):
+    """
+    An action node that uses a causal language model to generate
+    some text based on a prompt contained in the node's 
+    blackboard.
+
+    This node is based on the Hugging Face transformers library, and will
+    download the model that you specify by name. This can take a long 
+    time and/or use a lot of storage, depending on the model you name.
+
+    There are enough configuration options for this type of node that
+    the options have all been placed in a dataclass config object. See 
+    the documentation for that object to learn about the many options
+    available to you.
+
+    Args:
+        name (str):
+            The given name of this node.
+        cfg (CausalLMActionConfig):
+            The configuration object for this model.
+    """
+    def __init__(self, name : str, cfg : HFLMActionConfig) -> None:
+        super().__init__(name)
+
+        self.input_key = cfg.input_key
+        self.output_key = cfg.output_key
+        self.device = cfg.device
+        self.max_new_tokens = cfg.max_new_tokens
+        self.torch_dtype = cfg.dtype
+        self.temperature = cfg.temperature
+
+        if self.temperature == 0.0: 
+            self.do_sample = False
+        else:
+            self.do_sample = True
+
+        self.model = HFLM(model=cfg.model, device=self.device, parallelize=cfg.parallelize)
+
+        self.input_processor = None
+        self.output_processor = None
+
+    def set_model(self, new_model) -> None:
```

Comment:
@RichardKelley what should the type of new_model be? 

you didn't have it typed before, so maybe it's not important.

Reply:
I think it's ok to leave out the type hint here - it could be hflm.LM at the moment, but it's not critical to add.